# **任意の文字列を先頭にもつ電子掲示板のトリップを生成するプログラム**

## **概要**

任意の文字列を先頭にもつ電子掲示板のトリップを生成するプログラムです。



## **操作方法**

実行すると先頭に表示させたい文字列の入力を求められます。
そこに表示させたい文字列を入力してください。<br>
カンマ区切りで複数入力し、そのうちのどれか1つを先頭に持つようにすることもできます。
### 例
```
ABC
ABC,DEF,123
```
4文字以上だと、かなり時間がかかります。


## **制作者のコメント**

ローカル環境での実行を試みたところ、Windowsでは動かないようです。<br>
Linuxでは動きました。

## **作成時間**

作成時間：**約9時間**

この課題に関連して行なった調査や学習に要した時間も含めてた時間を記述してください。

In [1]:
%reset -f
from crypt import crypt
x = False
#入力された文字列を分割############
str_s = str(input("先頭に表示させたい文字列:"))
list_s = str_s.split(",")
for tripkey in range(10000000000):
  if x == True:
      break
#トリップ生成######################
  tripkey = str(tripkey)
  salt = (tripkey + "H.")[1:3]
  trip = crypt(tripkey, salt)
  trip = trip[-10:]
#比較##############################
  for s in list_s:
    if x == True:
      break
    if s == trip[0:len(s)]:
      print(f"trip: ◆{trip}")
      print(f"key: #{tripkey}")
      x = True
      break
if x == False:
  print("見つかりませんでした。")

先頭に表示させたい文字列:test,TEST,Test
trip: ◆TestrA8IMA
key: #27792091


## **結果のメモ**

先頭に表示させたい文字列:test,TEST,Test<br>
trip: ◆TestrA8IMA<br>
key: #27792091<br>

先頭に表示させたい文字列:AAAA,BBBB,CCCC,DDDD,EEEE,FFFF<br>
trip: ◆EEEEryJjWo<br>
key: #111256<br>

先頭に表示させたい文字列:00000,11111,22222,33333,44444,55555,66666,77777,88888,99999,aaaaa,bbbbb,ccccc,ddddd,eeeee,fffff,ggggg,hhhhh,iiiii,jjjjj,kkkkk,lllll,nnnnn,mmmmm,ooooo,ppppp,qqqqq,rrrrr,sssss,ttttt,uuuuu,vvvvv,wwwww,xxxxx,yyyyy,zzzzz,AAAAA,BBBBB,CCCCC,DDDDD,EEEEE,FFFFF,GGGGG,HHHHH,IIIII,JJJJJ,KKKKK,LLLLL,MMMMM,NNNNN,OOOOO,PPPPP,QQQQQ,RRRRR,SSSSS,TTTTT,UUUUU,VVVVV,WWWWW,XXXXX,YYYYY,ZZZZZ,.....,/////<br>
trip: ◆jjjjjnLmmg<br>
key: #21295790<br>

先頭に表示させたい文字列:00000,11111,22222,33333,44444,55555,66666,77777,88888,99999,AAAAA,BBBBB,CCCCC,DDDDD,EEEEE,FFFFF,GGGGG,HHHHH,IIIII,JJJJJ,KKKKK,LLLLL,MMMMM,NNNNN,OOOOO,PPPPP,QQQQQ,RRRRR,SSSSS,TTTTT,UUUUU,VVVVV,WWWWW,XXXXX,YYYYY,ZZZZZ<br>
trip: ◆TTTTTvkfjU<br>
key: #29221916<br>